x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/fulltext/MatchTests.java

```diff
@@ -108,10 +105,34 @@ private static List<TestCaseSupplier.TypedData> getTestParams(List<DataType> par
                 "field"
             )
         );
-        params.add(new TestCaseSupplier.TypedData(new BytesRef(randomAlphaOfLength(10)), paramDataTypes.get(1), "query"));
+        final Object value = randomQuery(paramDataTypes.get(1));
+        params.add(new TestCaseSupplier.TypedData(new BytesRef(String.valueOf(value)), paramDataTypes.get(1), "query"));
         return params;
     }
 
+    public static Object randomQuery(DataType dataType) {
+        if (Match.DATA_TYPES.contains(dataType) == false) {
+            throw new IllegalArgumentException("Unsupported type in tests: " + dataType);
+        }
+        Object value = EsqlTestUtils.randomLiteral(dataType).value();
+        if (value instanceof BytesRef bytesRef) {
```

Comment:
It was a bit tricky to get the correct base object from the `randomLiteral` function, but I wanted to reuse it for calculating random values for specific data types